### PR TITLE
WIP feat: intergrate opentelemetry tracing

### PR DESCRIPTION
**Pull request by fatelei**
ref: https://github.com/cectc/dbpack/issues/<issueID>

### Ⅰ. Describe what this PR did

#50 
- add opentelemetry support in dbpack

### CODE REVIEW

**Comment by fatelei** [comment]
this pr is wip

## PR detail
firstly, we add opentelemetry for dbpack, after this pr worked, the second pr will add trace info in sql comment

**Comment by dk-lockdown** [comment]
> this pr is wip
> 
> ## PR detail
> firstly, we add opentelemetry for dbpack, after this pr worked, the second pr will add trace info in sql comment

please fix ci errors

**Review by dk-lockdown — APPROVED** [review]
LGTM

**Comment by adofsauron** [comment]
LGTM

File: pkg/executor/read_write_splitting.go [hunk 1]
```diff
@@ -21,6 +21,8 @@ import (
 	"encoding/json"
 	"sync"
 
+	"github.com/cectc/dbpack/pkg/tracing"
```
Comment (bohehe) [inline pkg/executor/read_write_splitting.go:24]: Please format import.

File: pkg/executor/read_write_splitting.go [hunk 2]
```diff
@@ -214,26 +218,28 @@ func (executor *ReadWriteSplittingExecutor) ExecutorComQuery(ctx context.Context
...
 		if ok {
 			tx = txi.(proto.Tx)
-			return tx.Query(ctx, sql)
+			return tx.Query(newCtx, sql)
 		}
-		db = executor.masters.Next(proto.WithMaster(ctx)).(*DataSourceBrief)
-		return db.DB.Query(proto.WithMaster(ctx), sql)
+		withMasterCtx := proto.WithMaster(newCtx)
+		db = executor.masters.Next(withMasterCtx).(*DataSourceBrief)
+		return db.DB.Query(withMasterCtx, sql)
```
Comment (bohehe) [inline pkg/executor/read_write_splitting.go:233]: Place a line break before return

File: pkg/executor/sharding.go [hunk 3]
```diff
@@ -21,6 +21,8 @@ import (
 	"encoding/json"
 	"fmt"
 
+	"github.com/cectc/dbpack/pkg/tracing"
```
Comment (bohehe) [inline pkg/executor/sharding.go:24]: should sort import.

**Review by bohehe — APPROVED** [review]

**Pull request merged**